lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -100,29 +104,36 @@ def get_course_topics(self, user=None):
         Get course topics for self.course, using the given user or self.user if
         not provided, and generating absolute URIs with a test scheme/host.
         """
-        return get_course_topics(self.course.id, user or self.user)
+        if user:
+            self.request.user = user
+        return get_course_topics(self.request, self.course.id)
 
-    def make_expected_tree(self, topic_id, name, children=None):
+    def make_expected_tree(self, topic_id_list, name, children=None):
         """
-        Build an expected result tree given a topic id, display name, and
-        children
+        Build an expected result tree given a topic id, display name, children
+        and thread_list_url
         """
         children = children or []
         node = {
-            "id": topic_id,
             "name": name,
             "children": children,
+            "thread_list_url": get_thread_list_url(self.request, self.course.id, topic_id_list)
```

Comment:
This shouldn't call the code under test, because that makes it too easy to make a breaking change in that function without causing tests to fail.